gpytorch/test/lazy_tensor_test_case.py

```diff
@@ -578,3 +578,29 @@ def test_sqrt_inv_matmul_no_lhs(self):
         for arg, arg_copy in zip(lazy_tensor.representation(), lazy_tensor_copy.representation()):
             if arg_copy.grad is not None:
                 self.assertAllClose(arg.grad, arg_copy.grad, rtol=1e-4, atol=1e-3)
+
+    def test_symeig(self):
+        lazy_tensor = self.create_lazy_tensor().requires_grad_(True)
+        lazy_tensor_copy = lazy_tensor.clone().detach_().requires_grad_(True)
+        evaluated = self.evaluate_lazy_tensor(lazy_tensor_copy)
+
+        # Perform forward pass
+        evals, evecs = lazy_tensor.symeig(eigenvectors=True)
+        evecs = evecs.evaluate()
+
+        # since LazyTensor.symeig does not sort evals, we do this here for the check
+        evals_sorted, idxr = torch.sort(evals, dim=-1, descending=False)
+        evecs_sorted = torch.gather(evecs, dim=-1, index=idxr.unsqueeze(-2).expand(evecs.shape))
```

Comment:
this is kind of dense, torch should really have a more convenient batched indexing functionality....